src/js/reducers/expandedLayers.js

```diff
@@ -0,0 +1,29 @@
+/* eslint new-cap:0 */
+'use strict';
+
+var assign = require('object-assign');
+var Immutable = require('immutable');
+
+function selectedMarkReducer(state, action) {
+  var newState;
+  if (typeof state === 'undefined') {
+    return Immutable.Map();
+  }
+  if (action.type === 'EXPAND_LAYERS') {
+    newState = action.layerIds.reduce(function(layers, layerId) {
+      layers[layerId] = true;
+      return layers;
+    }, {});
+    return assign({}, state, newState);
+  }
```

Comment:
The key thing I need help with is that I couldn't get these to work by using Immutable methods; the initial state should be an immutable map, but any subsequent change creates a werid, not-quite-Immutable object. `return state.merge()` did not work the way I expected.